cmake/fckit_preprocess_fypp.cmake

```diff
@@ -20,27 +20,6 @@ function( fckit_target_append_fypp_args output target )
       set( valid_target FALSE )
     endif()
   endif()
-  if( valid_target )
-    if( CMAKE_VERSION VERSION_LESS 3.12 ) # Hopefully we can remove this soon
-      foreach( include_property INCLUDE_DIRECTORIES;INTERFACE_INCLUDE_DIRECTORIES )
-        set( prop "$<TARGET_PROPERTY:${target},${include_property}>" )
-        list( APPEND _args "$<$<BOOL:${prop}>:-I $<JOIN:${prop}, -I >>" )
-      endforeach()
-      foreach( definitions_property COMPILE_DEFINITIONS;INTERFACE_COMPILE_DEFINITIONS )
-        set( prop "$<TARGET_PROPERTY:${target},${definitions_property}>" )
-        list( APPEND _args "$<$<BOOL:${prop}>:-D $<JOIN:${prop}, -D >>" )
-      endforeach()
-    else()
-      foreach( include_property INCLUDE_DIRECTORIES;INTERFACE_INCLUDE_DIRECTORIES )
-        set( prop "$<$<TARGET_EXISTS:${target}>:$<TARGET_PROPERTY:${target},${include_property}>>" )
-        list( APPEND _args "$<$<BOOL:${prop}>:-I $<JOIN:${prop}, -I >>" )
-      endforeach()
-      foreach( definitions_property COMPILE_DEFINITIONS;INTERFACE_COMPILE_DEFINITIONS )
-        set( prop "$<$<TARGET_EXISTS:${target}>:$<TARGET_PROPERTY:${target},${definitions_property}>>" )
-        list( APPEND _args "$<$<BOOL:${prop}>:-D $<JOIN:${prop}, -D >>" )
-      endforeach()
-    endif()
-  endif()
```

Comment:
This code is required to automatically give the fypp preprocessor access to the target compile definitions and target include directories.
What is the issue w.r.t. robustness here?